pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -852,6 +859,13 @@ func getInstallerPodName(ns *operatorv1.NodeStatus) string {
 
 // ensureInstallerPod creates the installer pod with the secrets required to if it does not exist already
 func (c *InstallerController) ensureInstallerPod(ctx context.Context, operatorSpec *operatorv1.StaticPodOperatorSpec, ns *operatorv1.NodeStatus) error {
+	// checks if new revision should be rolled out
+	if c.shouldRevisionInstall != nil {
+		shouldInstall, err := c.shouldRevisionInstall()
+		if !shouldInstall {
```

Comment:
shouldn't that be

```suggestion
		if err != nil {
    		   return err
		}
		if !shouldInstall {
		  return nil
		}
```

Reply:
maybe also add an info logging statement here, so we know if it was skipped

Reply:
@tjungblu But only if there is error the requeue  will happen as per the current logic here. https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/controller/installer/installer_controller.go#L481.

Reply:
then we need to rethink the ensureInstallerPod signature potentially. I don't think we should trigger an event for this for example:
```
c.eventRecorder.Warningf("InstallerPodFailed", "Failed to create installer pod for revision %d count %d on node %q: %v", 
       currNodeState.TargetRevision, currNodeState.LastFailedCount, currNodeState.NodeName, err) 
```

that's misleading, because there was no actual error that failed the installation. We just skipped it.